en/docs/reference/synapse-properties/synapse-expressions.md

```diff
@@ -0,0 +1,777 @@
+<td><pre><code>Transforms the input date/time from the specified old format to the new format.
+Accepts string inputs or results from <code>now()</code>.</code></pre></td>
+</tr>
+</tbody>
+</table>
+
+### Fetch secrets
+
+Synapse Expressions can be used to fetch secrets from the secure-vault.
+
+Syntax for fetching secrets
+```
+secret(alias)
+```
+
+Example usage:
+```
+<variable name="password" expression="${secret('mysqlpassword')}"/>
+```
+
+### Check exists function
+
+Synapse Expressions includes the following function to check if a property exists.
+
+Syntax for checking if a property exists
+```
+exists(value)
+```
+
+Example usage:
+```
+<variable name="var1" expression="${exists('var.notMandatoryProperty') ? var.notMandatoryProperty : 'default' }"/>
+```
+
+### Invert function
+
+Not function can be used to invert a boolean value.
+
+Syntax for inverting a boolean value
+```
+not(value)
+```
+
+Example usage:
+```
+<variable name="var1" expression="${not(payload.booleanValue)}"/>
+```
+
+### XPATH functions
+
+Synapse Expressions provide XPATH functions to work with XML payloads.
+
+Syntax for calling a XPATH functions without namespaces
+
+```
+xpath(expression)
+```
+
+Example usage: Calling a XPATH functions without namespaces
+```
+<variable name="StudentName" expression="${xpath('//student/text()')}"/>
+```
+
+Example usage: Calling a XPATH functions with namespaces
+```
+<variable name="Width" expression="${xpath('//a:parent/b:child/a:value/text()')}" xmlns:a="http://namespaceA.com" xmlns:b="http://namespaceB.com"/>
+```
+
+Syntax for calling a XPATH function on a variable
+
+```
+xpath(expression, 'variable-name')
+```
+
+Example usage:
+```
+<variable name="StudentName" expression="${xpath('//student/text()', 'Students')}"/>
+```
+
+## Reserved Keywords
+
+The following keywords are reserved in Synapse Expressions and cannot be used in value access expressions with dot notation (instead we should use the bracket notation).
+
+<table>
+<thead></thead>
+<tbody>
+<tr class="odd">
+<td><pre><code>contains</code></pre></td>
+<td><pre><code>in</code></pre></td>
+<td><pre><code>nin</code></pre></td>
+<td><pre><code>subsetof</code></pre></td>
+<td><pre><code>size</code></pre></td>
+</tr>
+<tr class="even">
+<td><pre><code>empty</code></pre></td>
+<td><pre><code>empty true</code></pre></td>
+<td><pre><code>empty false</code></pre></td>
+<td><pre><code>var</code></pre></td>
+<td><pre><code>payload</code></pre></td>
+</tr>
+<tr class="odd">
+<td><pre><code>headers</code></pre></td>
+<td><pre><code>config</code></pre></td>
+<td><pre><code>properties</code></pre></td>
+<td><pre><code>props</code></pre></td>
+<td><pre><code>params</code></pre></td>
+</tr>
+</tbody>
+</table>
+
+## Best practices
+
+SynapsePath provides a fault tolerance design which avoids disruptive failures. In case of an error in a complex expression it fails in the first evaluation without propagating the invalid result to rest of the evaluation steps
```

Comment:
```suggestion
SynapsePath provides a fault tolerance design which avoids disruptive failures. In case of an error in a complex expression, it fails in the first evaluation without propagating the invalid result to the rest of the evaluation steps.
```